### ISSUE TITLE
Connecting to XMPP (BOSH) server via browser using Strophe

### ISSUE THREAD

**Issue by markhorsman** [node-xmpp/node-xmpp-server#122]
Hi all,

I made a nodejs service for keeping track of presence statuses of my clients using a web application.
The client (browser) connects to the XMPP BOSH server via Strophe (client js library) passing in a custom session id which I use to authenticate the user via my mysql database. 
Once authenticated, the last known status is retrieved from a Redis storage and the status is send back to all active sessions connected to the server. I use the XMPP server to retrieve and send statuses.

This all worked fine when I was using v1.0.0-alpha12, but now I updated to 2.0.0 and the connection is not working anymore!!!

The authentication still works fine but the client never gets a response that its connected. After a few minutes the connection is automatically closed. 

These are the responses that I get back from my connection using Strophe:

```
<body xmlns="http://jabber.org/protocol/httpbind" xmlns:xmpp="urn:xmpp:xbosh" xmlns:stream="http://etherx.jabber.org/streams" sid="b07e300c9a726ff5e4a041fa4f05452c" xmpp:restartlogic="true"><features xmlns="http://etherx.jabber.org/streams"><mechanisms xmlns="urn:ietf:params:xml:ns:xmpp-sasl"><mechanism>PLAIN</mechanism></mechanisms></features></body>

<body xmlns="http://jabber.org/protocol/httpbind" xmlns:xmpp="urn:xmpp:xbosh" xmlns:stream="http://etherx.jabber.org/streams" sid="b07e300c9a726ff5e4a041fa4f05452c"><success xmlns="urn:ietf:params:xml:ns:xmpp-sasl"/></body>

<body xmlns="http://jabber.org/protocol/httpbind" xmlns:xmpp="urn:xmpp:xbosh" xmlns:stream="http://etherx.jabber.org/streams" sid="b07e300c9a726ff5e4a041fa4f05452c"><features xmlns="http://etherx.jabber.org/streams"><bind xmlns="urn:ietf:params:xml:ns:xmpp-bind"/><session xmlns="urn:ietf:params:xml:ns:xmpp-session"/></features></body>
```

Says nothing to me!

I'm using the latest stable version of Node (4.2.1) and running my app on Ubuntu 14.01.

Is there maybe an example of the scenario I just described? Meaning I want to connect from client side (browser) to my XMPP server (using BOSH or something else that works!) an establishing an connection (stream).

I hope you can help me with this problem!

Thanks!

**Comment by sonnyp** [comment]
Looks like a regression, unfortunately BOSH support on node-xmpp-server is not maintained.
Good news is you can use WebSocket instead which works great both on Strophe and node-xmpp-server.

**Comment by markhorsman** [comment]
Thanks for your quick reply. Do you have an example of the WebSocket technique together with node-xmpp-server?

**Comment by markhorsman** [comment]
I got it working using WebSocket instead of BOSH locally. 

On my testing server I have an https connection on the browser and I try to connect to my xmpp server through WebSocket using wss.
This unfortunately does not work. Are there any options I have to pass to WebSocketServer object?

When I was using BOSH I passed in secure: true, key and cert files. When I try this with WebSocketServer this has no effect.

I'm almost there so I hope anyone knows what I have to do to get wss working.

Thanks!
